### ISSUE TITLE
Apple Signature no longer detected with new certificate

### ISSUE THREAD

**Issue by nebojsac** [tomasmcguinness/pkpassvalidator#15]
For some reason the validation fails (all the "Signature" bullet points are red) but the generated pass is valid (it gets added to the Wallet just fine).

This happens when we use a newly generated Pass Type certificate. It still works fine with the old Pass Type certificate, but that one is expiring soon.

**Comment by safonovklim** [comment]
@nebojsac which certificate do you use?

<img width="501" alt="image" src="https://user-images.githubusercontent.com/9365521/154867879-3424ae96-9673-4d2b-b21e-6f086a42f8de.png">

**Comment by nebojsac** [comment]
@safonovklim  - from that list - the first one.
But the thing is, when I use it with our old "Pass Type ID" certificate that you generate in your Apple Dev account, it validates fine. It fails to validate when everything else (including the WDR cert) is the same, the only difference being the newly generated "Pass Type ID" certificate. I had generated a few of them with different CSRs (Certificate Signing Requests) but I hit the account limit and I can't generate new ones for more tests.

**Comment by tomasmcguinness** [comment]
To the bet of my knowledge, you must use the G1.

This might be relevant

<img width="633" alt="image" src="https://user-images.githubusercontent.com/302741/154949964-57f1bfcf-5739-4819-9a0c-a063724d5c73.png">

from https://developer.apple.com/support/expiration/

**Comment by tomasmcguinness** [comment]
It's been the G1 certificate for a long time.

Can you check the certificate chain of the newly generated pass type certificate and see what the expiration date of the root cert is?

It's possible Apple have changed the version of the WWDR cert they are using.

**Comment by safonovklim** [comment]
Hey! Thank yall for followup!

I found that there was an issue with download/export process on my side. WWDR G1 is still valid even if your own cert is issued by WWDR CA G3 or G4.

just for reference - explanation: https://github.com/alexandercerutti/passkit-generator/issues/105

**Comment by tomasmcguinness** [comment]
Interesting. Thanks. It was only a matter of time before Apple started signing the Passkit certificates with something other than the G1 cert. I'll try and find some time this week to update the validator.

**Comment by alexandercerutti** [comment]
Hey @tomasmcguinness, I found an expiration notice: https://developer.apple.com/support/expiration/

Especially, it is written that:

> Apple Push Notification Service SSL Certificates, **including the Pass Type ID Certificate**, were updated January 27, 2022, and associated with a new intermediate certificate focused on the Apple Push Notification service (APNs). If you plan to send notification requests or passes with certificates issued after January 27, 2022, **you’ll need to [download](https://www.apple.com/certificateauthority/) the Worldwide Developer Relations G4 sub CA**, available for download. You can confirm that the correct intermediate certificate is installed by verifying that the expiration date is set to 2030 and the Organization field is set to G4.

I'll proceed in performing more checks for what concerns passkit-generator and updating Wiki instruction as soon I'll have more information.

**Comment by vpier** [comment]
Hi @tomasmcguinness & @alexandercerutti 

Amazing work from both of you, kudos for all your contributions.

@nebojsac I have pretty much the same issue with the validations performed, i.e. I have a valid pkpass that loads fine without any issues in Mac console, simulator and different devices; however pkpassvalidator reports issues with:
1. Signature (all checks)
2. WWDR Certificate Subject
3. PassKit Certificate has an incorrect value <- although I've followed the advice to leave common name field empty during CSR generation
4.  PassKit Certificate in date

The pass is generated with WWDR G4 cert. If I use WWDR G1 cert then the issues reported by pkpassvalidator are different, i.e.:
1. Signature (all checks)
2. PassKit Certificate has an incorrect value <- although I've followed the advice to leave common name field empty during CSR generation
3.  PassKit Certificate in date

Any ideas are more than welcomed :-)
